catroid/src/androidTest/java/org/catrobat/catroid/test/catblocks/WebViewUtils.kt

```diff
@@ -0,0 +1,393 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2023 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.test.catblocks
+
+import android.app.Activity
+import android.util.Log
+import android.webkit.JavascriptInterface
+import android.webkit.WebView
+import com.badlogic.gdx.math.Rectangle
+
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import org.catrobat.catroid.R
+import org.json.JSONObject
+
+class WebViewUtils(private var activity: Activity, timeoutSeconds: Long? = null) {
+    private val DEFAULT_TIMEOUT_SECONDS: Long = 5
+    private var TIMEOUT_SECONDS: Long = DEFAULT_TIMEOUT_SECONDS
+    private var webView: WebView
+    private val jsInterface: JSInterface
+    private val pageLoadLatch: CountDownLatch
+
+    init {
+        this.webView = activity.findViewById(R.id.catblocksWebView)
+        if (timeoutSeconds != null) {
+            TIMEOUT_SECONDS = timeoutSeconds
+        }
+
+        pageLoadLatch = CountDownLatch(1)
+        jsInterface = JSInterface(pageLoadLatch)
+        activity.runOnUiThread {
+            webView.addJavascriptInterface(jsInterface, "webViewUtils")
+            webView.loadUrl("https://appassets.androidplatform.net/assets/catblocks/index.html")
+        }
+    }
+
+    private fun getIsVisibleFunction(): String {
+        return """
```

Comment:
Wouldn't it be better to just define the entirety of the JS code in a seperate file?

Then evaluate this at the start of the tests
And then just call into the functions / evaluate them where necessary